app/components/amount-input-screen/amount-input-screen.tsx

```diff
@@ -198,11 +200,32 @@ export const AmountInputScreen: React.FC<AmountInputScreenProps> = ({
     [currencyInfo],
   )
 
-  const onToggleCurrency =
-    secondaryNewAmount &&
-    (() => {
+  const client = useApolloClient()
+  const swapDefaultUnitOfAccount =
```

Comment:
I think this implementation may have a bug. Its not a given that the primary amount is always the display currency or always the settlement currency. If swap was set to true and one use of the component passes in a display amount and another use of the component passes in a settlement amount, they would both end up swapping . A more accurate name would be `useDisplayAmountAsInitialAmount` and then you could explicitly handle the amount currencies before they reach this component.

Reply:
in what scenario would we not have a display currency or settlement currency? ie: what would make it that the order would be different?

Reply:
If on receive screen the initial primary amount was set to `0 Display` as the default and if on the send flow the primary amount was set to `0 Settlement` as the default. This isn't what we do now, but we dont have anything in place to prevent someone from misusing the component. 

Right now I believe we leave the amounts as undefined and then the initial amount is set in this line https://github.com/GaloyMoney/galoy-mobile/blob/c7c81a1f9d2df633e76752b964ab1a82555c3041/app/components/amount-input-screen/amount-input-screen.tsx#L147 with the `zeroDisplayAmount`. We could get rid of `needSwapCurrencyInit` and in this line use the value `useDisplayAmountAsDefaultAmount` to decide between `zeroDisplayAmount` and `zeroSettlementAmount` as the default.